src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+  const clear = 'clear';
+  const res = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case add:
+        Object.assign(stateCopy, action.extraData);
+        break;
+      case remove:
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+      case clear:
+        for (const prop in stateCopy) {
+          delete stateCopy[prop];
+        }
+        break;
+      default:
+        break;
```

Comment:
It would be better to inform user that something went wrong
```suggestion
      default:
        throw new Error("Unsupported action type");
```